lib/doh.c

```diff
@@ -407,6 +441,29 @@ struct Curl_addrinfo *Curl_doh(struct Curl_easy *data,
     dohp->pending++;
   }
 #endif
+
+#ifdef USE_HTTPSRR
+  if(port == 443) {
+    qname = strdup(hostname);
+  }
+  else {
+   /* the 16 is the most we get with a 5-digit port */
+   size_t qnamelen = strlen(hostname) + 16;
+
+    qname = (char *)calloc(1, qnamelen);
+    if(!qname)
+      goto error;
+    curl_msnprintf(qname, qnamelen, "_%d._https.%s",
+                     port % 0xffff, hostname);
```

Comment:
The port number will be legit, no need to `% 0xffff` it. Which is wrong anyway since 0xffff is a legitimate port number.

Also: I prefer using `aprintf()` directly for things like this instead of calloc + msnprintf, as then you don't have to calculate the max space like this.

Reply:
did that, thanks